packages/flutter/lib/src/material/data_table.dart

```diff
@@ -85,6 +86,13 @@ class DataColumn {
   final DataColumnSortCallback? onSort;
 
   bool get _debugInteractive => onSort != null;
+
+  /// The cursor for a mouse pointer when it enters or is hovering over the
+  /// heading row.
+  ///
+  /// If this is null, then the value of [DataTableThemeData.headingCellCursor]
+  /// is used. If that's null, then [SystemMouseCursors.click] is used.
```

Comment:
Is using [SystemMouseCursors.click] changing the current default behavior? Is [SystemMouseCursors.click] the right default for all platforms? If the user can click on a header, do we need to use a MaterialStateProperty here so that we can support pressed and disabled states too?

Reply:
> do we need to use a MaterialStateProperty here so that we can support pressed and disabled states too?

Yes, we can, since the header wraps itself in `inkWell` even if the header is disabled but there is a [TODO](https://github.com/flutter/flutter/blob/250a6c294264d3e8b39b5580737f2ed959d1de5a/packages/flutter/lib/src/material/data_table.dart#L836-L844) to do the opposite and not wrap the header with inkwell when `onSort` is disabled.
Just added `MaterialStateProperty` for the header cell.

However, the row doesn't wrap itself when disabled so there is no way to get a disabled mouse cursor.
https://github.com/flutter/flutter/blob/250a6c294264d3e8b39b5580737f2ed959d1de5a/packages/flutter/lib/src/material/data_table.dart#L910-L919